### PR TITLE
Add lutro.joystickpressed() callback

### PR DESCRIPTION
**Pull request by RobLoach**
Add `lutro.joystickpressed()` and `lutro.joystickreleased()` events.

### CODE REVIEW

**Comment by twinaphex** [comment]
Can we get some more input on this? Maybe also invite @Kivutar

File: joystick.c [hunk 1]
```diff
@@ -79,3 +119,105 @@ int joystick_isDown(lua_State *L)
...
+/**
+ * Retrieve the Joystick string from a given libretro enum key.
+ *
+ * @return The string if the key is found, an empty string otherwise.
+ */
+char* joystick_retroToJoystick(int joystickKey)
+{
+  switch (joystickKey) {
+    case RETRO_DEVICE_ID_JOYPAD_B:
+      return "b";
```
Comment (RobLoach) [inline joystick.c]: Rather then this switch statement, we should use a dictionary.

**Comment by RobLoach** [comment]
Do you think we want to invoke the gamepad events instead?
https://love2d.org/wiki/love.gamepadpressed

**Comment by Kivutar** [comment]
Can I merge this @RobLoach ?

**Comment by RobLoach** [comment]
Sure! :+1:

**Pull request merged**